src/ucp/proto/proto_common.c

```diff
@@ -379,8 +379,15 @@ ucp_proto_common_get_lane_perf(const ucp_proto_common_init_params_t *params,
     ucp_proto_perf_node_own_child(perf_node, &lane_perf_node);
 
     /* For zero copy send, consider local system topology distance */
-    if (params->flags & UCP_PROTO_COMMON_INIT_FLAG_SEND_ZCOPY) {
-        sys_dev = params->super.select_param->sys_dev;
+    if (params->reg_mem_info.type != UCS_MEMORY_TYPE_UNKNOWN) {
```

Comment:
why need to check mem type?
also, do we need to update comment in line 381?

Reply:
If reg_mem_info type is not unknown we assume the protocol is going to send that mem type in a zero copy fashion. So, need to consider the system distance